### PR TITLE
Adding fuzz tests

### PR DESCRIPTION
**Pull request by jferas**
Adding fuzz tests for FranchiserFactory and Franchiser

### CODE REVIEW

File: test/FranchiserFactory.t.sol [hunk 1]
```diff
@@ -94,15 +79,62 @@ contract FranchiserFactoryTest is
...
+        franchiserFactory.fund(_delegatee, _amount);
+        vm.stopPrank();
+    }
+
+    function testFuzz_FundFailsWhenBalanceTooLow(address _delegator, address _delegatee, uint256 _amount) public {
+        vm.assume(_validActorAddress(_delegator));
+        vm.assume(_delegatee != address(0));
+        _amount = bound(_amount, 10, 100_000_000e18);
+
+        votingToken.mint(_delegator, _amount - 10);
```
Comment (alexkeating) [inline test/FranchiserFactory.t.sol]: The 10 could probably be fuzzed

File: test/FranchiserFactory.t.sol [hunk 2]
```diff
@@ -94,15 +79,62 @@ contract FranchiserFactoryTest is
...
+        votingToken.mint(_delegator, _amount);
+
+        vm.startPrank(_delegator);
+        votingToken.approve(address(franchiserFactory), _amount);
+        vm.expectRevert(IFranchiserErrors.NoDelegatee.selector);
+        franchiserFactory.fund(_delegatee, _amount);
+        vm.stopPrank();
+    }
+
+    function testFuzz_FundFailsWhenBalanceTooLow(address _delegator, address _delegatee, uint256 _amount) public {
```
Comment (alexkeating) [inline test/FranchiserFactory.t.sol]:
```suggestion
    function testFuzz_RevertIf_BalanceIsTooLow(address _delegator, address _delegatee, uint256 _amount) public {
```

File: test/FranchiserFactory.t.sol (hunk 1 above)
Comment (jferas) [reply test/FranchiserFactory.t.sol]: Done: 5cfccd88cfb2b5afe52ed304fcc32210437c9c1e

File: test/FranchiserFactory.t.sol (hunk 2 above)
Comment (jferas) [reply test/FranchiserFactory.t.sol]: There are many existing tests expressed with "Fails" instead of "RevertIf"

Comment (jferas) [reply test/FranchiserFactory.t.sol]: Done (for new tests): c380e785b3092ef7500dad3311dbef14c6855761

File: test/Franchiser.t.sol [hunk 3]
```diff
@@ -203,6 +211,55 @@ contract FranchiserTest is Test, IFranchiserErrors, IFranchiserEvents {
...
+        address _delegator,
+        address _delegatee,
+        address _subDelegatee,
+        uint256 _amount,
+        uint256 _delta
+    ) public {
+        vm.assume(_validActorAddress(_delegator));
+        vm.assume(_delegatee != address(0));
+        vm.assume(_subDelegatee != address(0));
+        vm.assume((_amount >= _delta) && (_amount <= 100_000_000e18));
```
Comment (alexkeating) [inline test/Franchiser.t.sol]: Is this necessary with the bound statements below

Comment (jferas) [reply test/Franchiser.t.sol]: Good catch.. not needed.. thanks! 77bf13e772becab7b48d4a0f10297b36058d5f63

**Review by alexkeating — APPROVED** [review]

**Pull request merged**